tests/unit/mathematics/linalg/SGLinalg_unittest.cc

```diff
@@ -0,0 +1,39 @@
+#include <shogun/lib/config.h>
+
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/mathematics/linalg/LinalgBackendViennaCL.h>
+#include <shogun/lib/SGVector.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+
+TEST(LinalgRefactor, CPU_dot)
```

Comment:
the test name is meaningless, in the test you do vienna, doesnt match. please clean up